tpm2/test/tpm2_test.go

```diff
@@ -575,6 +575,144 @@ func TestCertify(t *testing.T) {
 	})
 }
 
+func TestCertifyEx(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	params := Public{
+		Type:       AlgRSA,
+		NameAlg:    AlgSHA256,
+		Attributes: FlagSignerDefault,
+		RSAParameters: &RSAParams{
+			Sign: &SigScheme{
+				Alg:  AlgRSASSA,
+				Hash: AlgSHA256,
+			},
+			KeyBits: 2048,
+		},
+	}
+	signerHandle, signerPub, err := CreatePrimary(rw, HandleOwner, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(signer) failed: %s", err)
+	}
+	defer FlushContext(rw, signerHandle)
+
+	subjectHandle, subjectPub, err := CreatePrimary(rw, HandlePlatform, pcrSelection7, emptyPassword, defaultPassword, params)
+	if err != nil {
+		t.Fatalf("CreatePrimary(subject) failed: %s", err)
+	}
+	defer FlushContext(rw, subjectHandle)
+
+	attest, sig, err := CertifyEx(rw, defaultPassword, defaultPassword, subjectHandle, signerHandle, nil, SigScheme{Alg: AlgRSASSA, Hash: AlgSHA256, Count: 0})
+	if err != nil {
+		t.Errorf("Certify failed: %s", err)
+		return
+	}
+
+	attestHash := sha256.Sum256(attest)
+	if err := rsa.VerifyPKCS1v15(signerPub.(*rsa.PublicKey), crypto.SHA256, attestHash[:], sig); err != nil {
+		t.Errorf("Signature verification failed: %v", err)
+	}
+
+	t.Run("DecodeAttestationData", func(t *testing.T) {
```

Comment:
This is awesome, but I don't think it's necessary to test that the TPM properly signed the right thing for the Certify. I think it's sufficient to verify that the TPM returned some data that was signed according to the scheme that was passed in (for each scheme that we pass in)

Reply:
I made sure `TestCertifyEx` replicates the same code that is in `TestCertify`, that's why this test is there. Are you sure you want this piece of code gone ?

Reply:
That's my thinking, @twitchy-jsonp or @josephlr what do you think?

Philosophically: this library is for making it easy to talk to the TPM. The unit test for this library should ensure that it successfully performs the intended action with the TPM, not that the TPM implementation itself is correct. IMHO, this code from the original Certify test makes it harder to read/maintain (for example, to add another iteration where we sign with another padding scheme)